### PR TITLE
Add in custom glib patch

### PR DESCRIPTION
**Pull request by andrewkatson**
@jsharpe So I was able to add in glib but not exactly sure where to point it to. I see in `MODULE.bazel` that we pull in `glib_src` and `glib_dev`. And I followed that path to `extensions.bzl` but I am not quite sure what is going on. Can you give me some hints?

### CODE REVIEW

**Comment by jsharpe** [comment]
AFAIK pkg-config bundles a vendored copy of glib -you'll need to add glib as a dependency to the pkg-config build target in the toolchains directory and then update the configure comand to point at that version of glib?
The existing glib_src and glib_dev I think are just in the examples if I remember correctly so those may be red herrings.

File: WORKSPACE.bazel [hunk 1]
```diff
@@ -72,3 +72,10 @@ load("@bazel_skylib_gazelle_plugin//:setup.bzl", "bazel_skylib_gazelle_plugin_se
 bazel_skylib_gazelle_plugin_setup()
 
 gazelle_dependencies(go_repository_default_config = "//:WORKSPACE.bazel")
+
+http_archive(
+    name = "glib",
+    urls = ["https://gitlab.gnome.org/GNOME/glib/-/archive/2.80.3/glib-2.80.3.zip"],
+    strip_prefix = "glib-2.80.3",
+    patches = ["https://lists.freedesktop.org/archives/pkg-config/attachments/20240503/7e621a18/attachment.bin"],
```
Comment (jsharpe) [inline WORKSPACE.bazel:80]: Its probably a good idea to mirror this into the repo as I assume this file isn't that big and relying on a mail archive might not be the most reliable.

**Comment by andrewkatson** [comment]
> AFAIK pkg-config bundles a vendored copy of glib -you'll need to add glib as a dependency to the pkg-config build target in the toolchains directory and then update the configure comand to point at that version of glib? The existing glib_src and glib_dev I think are just in the examples if I remember correctly so those may be red herrings.

Somewhere here? https://github.com/bazelbuild/rules_foreign_cc/blob/37383aaeba8dfbdbf2b0f765c2102c3963d2c8ab/toolchains/built_toolchains.bzl#L271

File: WORKSPACE.bazel (hunk 1 above)
Comment (andrewkatson) [reply WORKSPACE.bazel:80]: Okay Ive got it mirrored locally and patch is applied. Are there any files I should be excluding so I dont upload them?

**Comment by jsharpe** [comment]
See: https://github.com/bazelbuild/rules_foreign_cc/issues/1200#issuecomment-2166538237 - the patch linked here should be sufficient to fix this - @andrewkatson if you could turn that into a PR against this repo then that'd be great!

**Comment by andrewkatson** [comment]
Okay Ill do that

**Comment by andrewkatson** [comment]
@jsharpe I have a question. You said to mirror glib but whenever I try to mirror it I get prompted for their ssh password and then cannot. I've tried a few different ways of mirroring. Is it sufficient to just copy code from their latest release then apply the patch to that?

**Comment by jsharpe** [comment]
I actually don't think we need to pull in an updated glib looking at the previous patch I linked from the other issue - seems that pkg-config can be patched to fix this without needing to bring in a newer glib version. - you should be able to take in the patch that was linked, apply to the rules_foreign_cc and then fixup the Label references to not reference the repo that patch comes from and that should be sufficient.

**Comment by andrewkatson** [comment]
Sorry for all the questions I have never even heard of patching before this! @jsharpe I applied one of the patches. The other one talked about `.bazelrc` which I didnt see in the repo.

**Comment by andrewkatson** [comment]
Also tried to reproduce the errors above using the commands from the remote server and it just hangs trying to build the cmake toolchain. I am still trying to debug but if you can tell what the issue is from the buildkite checks then let me know

**Comment by andrewkatson** [comment]
@jsharpe thank you for that merge! Everything passes now! When can we merge it?

**Comment by barbu110** [comment]
Hey folks, just want to mention this is very important for our project - it's currently blocking builds. Trying to understand if there's a way to get this patch applied from our own project (that is, until this PR gets merged). Any tips?

**Review by jsharpe — APPROVED** [review]
LGTM - we're not currently checking in the MODULE.bazel.lock file so I've removed that.

**Comment by andrewkatson** [comment]
> Hey folks, just want to mention this is very important for our project - it's currently blocking builds. Trying to understand if there's a way to get this patch applied from our own project (that is, until this PR gets merged). Any tips?

Follow this https://aur.archlinux.org/packages/envoyproxy from https://github.com/bazelbuild/rules_foreign_cc/issues/1200#issuecomment-2166538237. Does that help?

**Comment by andrewkatson** [comment]
@jsharpe when do you think you'll publish a new version to the bazel central repo?

**Pull request merged**